### DISCUSSION TITLE
Multiple application detection rules

### DISCUSSION THREAD

**Discussion by mdicss** [Dynatrace/dynatrace-configuration-as-code#293] [Q&A] [answered]
Hi
Is it really not possible to define multiple application detection rules for one application with monaco? See: https://github.com/dynatrace-oss/dynatrace-monitoring-as-code/issues/254

To me, this seems a problem in the design of dynatrace here. In dynatrace appmon, you had the possibility to define one application together with a whole list of patterns (=detection rules). Now, in the GUI in the settings, we see only all the detection rules, not the applications themselves and not for every application, all rules belonging to this application.

**Comment by didiladi** [comment, marked as answer]
Yes, it is possible. Simply, add a `name` property into the json and define the name of the application detection config yourself. Then the API will not fallback to the mapped application anymore. Currently, when you download the config from the API (when defined it manually in the UI), it will not contain the property.

**Reply by mdicss** [reply]
Thank you for the answer, but sorry, I still cann't make it work. Could you give some more explanations or an example please?

I have a application-web configuration to define 1 application. Then I try to define two app-detection-rules for this application. In the json for the application-detection, I need a `name` and an `applicationIdentifier` property. I can put the name directly in the json of the app-detection-rule, but I still need the id and always get an error like 'Given application detection rule duplicates the name of rule...' or I end up with only one application detection rule instead of two.
Here the configuration I tried last:
application-web/application-web.yaml:
<pre>config:
- application: application-test.json
application:
- name: "application name"
</pre>
app-detection-rule/app-detection-rule.yaml:
<pre>config:
- app1: app1-test.json
- app2: app2-test.json

app1:
- name: "application-web/application.name"
- applicationId: "application-web/application.id"
- pattern: "foo"

app2:
- name: "gugus"
- applicationId: "application-web/application.id"
- pattern: "bar"</pre>
app-detection-rule/app1-test.json:
<pre>{
 "applicationIdentifier": "{{ .applicationId }}",
 "name": "{{.name}}",
 "filterConfig": {
  "applicationMatchTarget": "DOMAIN",
  "applicationMatchType": "BEGINS_WITH",
  "pattern": "{{ .pattern }}"
 },
 "metadata": {
  "configurationVersions": [
   0
  ]
 }
}</pre>
app-detection-rule/app2-test.json:
<pre>{
 "applicationIdentifier": "{{ .applicationId }}",
 "name": "application name",
 "filterConfig": {
  "applicationMatchTarget": "DOMAIN",
  "applicationMatchType": "BEGINS_WITH",
  "pattern": "{{ .pattern }}"
 },
 "metadata": {
  "configurationVersions": [
   0
  ]
 }
}</pre>

**Reply by didiladi** [reply]
Try changing the `name` in `app-detection-rule/app2-test.json` also to `{{ .name }}`

**Reply by mdicss** [reply]
Bingo! Thank you a lot.
But the solution isn't really obvious?

**Reply by didiladi** [reply]
You are absolutely right! We still need to extend the documentation and bring the feedback back to the engineering team.

**Comment by mdicss** [comment]
There remains a problem. When I deploy the same configuration a second time, I get an error about duplicate rules.
